internal/pkg/manifest/svc.go

```diff
@@ -172,23 +196,52 @@ func (c *Count) Desired() (*int, error) {
 // Percentage represents a valid percentage integer ranging from 0 to 100.
 type Percentage int
 
+// Resource represents a resource that has autoscaling configurations.
+type Resource struct {
```

Comment:
Can we make this struct name to be more expressive? How about `AdvancedScalingConfigOrPercentage`?